### PR TITLE
Next steps on DataPayload migration

### PR DESCRIPTION
**Pull request by sffc**

### CODE REVIEW

File: provider/core/src/data_provider.rs [hunk 1]
```diff
@@ -105,7 +104,74 @@ where
...
+        }
+    }
+
+    #[inline]
+    pub fn as_legacy_cow(&self) -> &Cow<'d, T> {
+        &self.cow
+    }
+
+    #[inline]
+    pub fn as_legacy_cow_mut(&mut self) -> &mut Cow<'d, T> {
```
Comment (Manishearth) [inline provider/core/src/data_provider.rs:137]: suggestion: next step is probably turning this into an `.as_mut(|&mut cow| {..})` closure-taking function, which we can support with Yoke

Comment (sffc) [reply provider/core/src/data_provider.rs]: Done.

**Comment by sffc** [comment]
I removed all external references to the `.cow`, including a call site using `.with_mut()`.

This is a reasonable stopping point for the afternoon; I'll continue later.  If you have a chance to review, that's fine; if not, I'll just continue working from the branch.

**Review by Manishearth — APPROVED** [review]
Looks pretty straightforward so far! I'm glad that `with_mut()` seems to cleanly work where we need it to.

**Pull request merged**